react/features/local-recording/recording/OggAdapter.js

```diff
@@ -0,0 +1,107 @@
+import { RecordingAdapter } from './RecordingAdapter';
+import { downloadBlob, timestampString } from './Utils';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * RecordingAdapter implementation that uses MediaRecorder
+ * (default browser encoding with Opus codec)
+ */
+export class OggAdapter extends RecordingAdapter {
+
+    _mediaRecorder = null;
+
+    /**
+     * Implements {@link RecordingAdapter#ensureInitialized()}.
+     *
+     * @inheritdoc
+     */
+    ensureInitialized() {
```

Comment:
Maybe call it "initialize" to be consistent with the rest of the codebase's use of singletons?

Reply:
The idea is that `ensureInitialized` can be called multiple times. If it is already initialized, it returns a resolved Promise, otherwise a pending promise.

Reply:
Okay, I see it now, the blatantly obvious line of `export class ...`. Why not initialize in the constructor?

Reply:
Okay, maybe you're not doing initialization in the constructor to avoid loading flac until it's needed.

Reply:
Initialization in the constructor has the same issue: the initialization is asynchronous (both the browser API and libflac). Unless we can "await" the constructor. Otherwise, there still needs to be a method that returns a Promise for ensuring initialization is complete.

Reply:
What I'm thinking then is having the init promise on the instance. Then in start it would be `initializationPromise.then(startRecordingLogic)`.

Reply:
So hiding it (the initialization process) from the caller essentially? Then I can remove that `ensureInitialized` method from the interface. Sounds good. 👍